internal/controller/virtualmachinebmc/controller.go

```diff
@@ -131,11 +201,55 @@ func (r *VirtualMachineBMCReconciler) constructServiceFromVirtualMachineBMC(virt
 	return svc
 }
 
-//+kubebuilder:rbac:groups=virtualmachine.kubevirt.io,resources=virtualmachinebmcs,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups=virtualmachine.kubevirt.io,resources=virtualmachinebmcs/status,verbs=get;update;patch
-//+kubebuilder:rbac:groups=virtualmachine.kubevirt.io,resources=virtualmachinebmcs/finalizers,verbs=update
-//+kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups=core,resources=services,verbs=get;list;watch;create;update;patch;delete
+func (r *VirtualMachineBMCReconciler) validateVirtualMachineExists(ctx context.Context, virtualMachineBMC *virtualmachinev1.VirtualMachineBMC) (bool, error) {
+	log := log.FromContext(ctx)
+
+	vmKey := client.ObjectKey{
+		Namespace: virtualMachineBMC.Namespace,
+		Name:      virtualMachineBMC.Spec.VirtualMachineRef.Name,
+	}
+
+	var vm kubevirtv1.VirtualMachine
+	if err := r.Get(ctx, vmKey, &vm); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("Referenced VirtualMachine not found",
+				"vm", virtualMachineBMC.Spec.VirtualMachineRef.Name,
+				"namespace", virtualMachineBMC.Namespace)
+
+			virtualMachineBMC.Status.Conditions = []metav1.Condition{
+				{
+					Type:   virtualmachinev1.ConditionReady,
+					Status: metav1.ConditionFalse,
+					Reason: "VirtualMachineNotFound",
+					Message: fmt.Sprintf("VirtualMachine %q not found in namespace %q",
+						virtualMachineBMC.Spec.VirtualMachineRef.Name,
+						virtualMachineBMC.Namespace),
+					LastTransitionTime: metav1.Now(),
+				},
+			}
```

Comment:
We can use the [`SetStatusCondition`](https://pkg.go.dev/k8s.io/apimachinery/pkg/api/meta#SetStatusCondition) to reduce the potential excessive status update calls:

```suggestion

			if changed := meta.SetStatusCondition(&cronJob.Status.Conditions, metav1.Condition{
				Type:    bmcv1.ConditionReady,
				Status:  metav1.ConditionFalse,
				Reason:  "VirtualMachineNotFound",
				Message: fmt.Sprintf("VirtualMachine %q not found in namespace %q",
						virtualMachineBMC.Spec.VirtualMachineRef.Name,
						virtualMachineBMC.Namespace),
			}); changed {
				return false, r.Status().Update(ctx, virtualMachineBMC)
			}
			return false, nil
```